### PR TITLE
Add Font Awesome Pro to the site [WIP]

### PR DESCRIPTION
**Pull request by levadadenys**
### **Add Font Awesome Pro to the site.**
Big update for Font Awesome: v4 -> v6 pro.
Hosting font awesome locally for pegasus/code.org,  pagasus/hourofcode.com sites, hosting font awesome from gem for dashboard pages

The main problem here is that in order download font-awesome-v6-pro gem we need to pass a secret token for our fontAwesomeKit. The way I propose to handle this is by passing this token inside bundler config by running
`bundle config dl.fontawesome.com token:SECRET_TOKEN` in your console from the project root directory.

However, I'm not sure how do we handle the case when some code.org student or teacher will wont to run/edit our sourcecode locally. Maybe we should get another token just for public usage (that we'll be able to suspend in case of strange token usage). Let me know what you think about that.

## Links
[Font Awesome Docs](https://fontawesome.com/docs/web/setup/use-kit)

Jira ticket: [SL-339](https://codedotorg.atlassian.net/browse/SL-339)

## Testing story
No tests added, just updating the usage of the library that already was installed.

## Deployment strategy
We need to update our .bundle/config file by running
`bundle config dl.fontawesome.com token:SECRET_TOKEN`

## Security
Handle the secret token correcty

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by kelbyhawn** [comment]
Hey @levadadenys, can you update the title and description up at the top to say what this PR is about?

**Comment by levadadenys** [comment]
> Hey @levadadenys, can you update the title and description up at the top to say what this PR is about?

Done, thanks for the hint! Basically it's a new version of font-awesome upgrade  pr [https://github.com/code-dot-org/code-dot-org/pull/49350](url)

**Comment by levadadenys** [comment]
The build is currently failing because of new lib added in branch, but it's absent in pr because of security issues for now

**Comment by kelbyhawn** [comment]
I think this looks good from the Pegasus side, but will wait to approve until @breville takes a look 🙌

File: pegasus/sites.v3/code.org/views/theme_common_head_before.haml [hunk 1]
```diff
@@ -10,6 +10,8 @@
 %link{rel:'apple-touch-icon', href:'/images/apple-touch-icon-precomposed.png'}
 
 %script{src:'https://www.googleoptimize.com/optimize.js?id=OPT-KBX3C3L'}
+-# Import FONT AWESOME PRO for all pegasus/sitev3/code.org pages
+%script{src: "https://kit.fontawesome.com/ea2d9d4413.js", crossorigin: "anonymous", async: true}
```
Comment (breville) [inline pegasus/sites.v3/code.org/views/theme_common_head_before.haml]: We should also make sure we aren't using a third-party domain for fonts on all pegasus sites as well.  The reason is that we want to make sure our site works in locations that have very strict firewalls, where they might have only permitted our domain but not other ones.

Comment (levadadenys) [reply pegasus/sites.v3/code.org/views/theme_common_head_before.haml]: issue solved)

File: pegasus/sites.v3/code.org/public/hourofcode2022.haml [hunk 2]
```diff
@@ -6,9 +6,6 @@ theme: responsive
 %link{href: "/css/generated/hoc2022-landing-page.css", rel: "stylesheet"}
 %link{href: "/css/hoc2022-nasa-skinny-banner.css", rel: "stylesheet"}
 
--# Import FontAwesome Pro
-%script{src: "https://kit.fontawesome.com/ea2d9d4413.js", crossorigin: "anonymous"}
-
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/public/hourofcode2022.haml:11]: @levadadenys did you check to make sure the icons were showing up on this page correctly w/ the local implementation?

Comment (levadadenys) [reply pegasus/sites.v3/code.org/public/hourofcode2022.haml:11]: Yep, everything works, looks like nothing is broken.
here's the fresh screenshots from this branch:
Знімок екрана 2022-12-14 о 21 44 27
Знімок екрана 2022-12-14 о 21 45 38
Знімок екрана 2022-12-14 о 21 45 42

Comment (kelbyhawn) [reply pegasus/sites.v3/code.org/public/hourofcode2022.haml:11]: Awesome, thanks for the screenshots!

**Comment by cat5inthecradle** [comment]
Is it necessary to add this many files to the repo? Can they be downloading during the build step? Passing a secret into the servers that build the project in our various environments is do-able.

Or perhaps we could host these files elsewhere. It just seems unnecessary to store this in our code repository when we ourselves will never be modifying these files or have a need for version control.

**Comment by levadadenys** [comment]
> Is it necessary to add this many files to the repo? Can they be downloading during the build step? Passing a secret into the servers that build the project in our various environments is do-able.
> 
> Or perhaps we could host these files elsewhere. It just seems unnecessary to store this in our code repository when we ourselves will never be modifying these files or have a need for version control.

Yeah, I'm not happy to ad as many files to repo as well.) The problem is that we need store these files localy for pegasus and exporter.js to work correctly.
I'm not sure there is a way to store them elsewhere, I've tried some options, but ended up following the previous structure. If you have any suggestions please let me know @breville @kelbyhawn  @cat5inthecradle

**Comment by breville** [comment]
In lieu of Git LFS, would S3 be a suitable place?

**Comment by cat5inthecradle** [comment]
For local, I would think we could make it part of other dependency installation steps and put the folder into the `.gitignore` file. It would mean that we need to share a secret with engineers to ensure they are able to download it.

S3 would work as well, I don't know if that's a practice we're doing elsewhere. Feels messy to me, but it might be fine.

Also, circling back to this note in the original PR description "how do we handle the case when some code.org student or teacher will wont to run/edit our sourcecode locally". That scenario doesn't happen, students and teachers aren't building and running our application locally. Engineers and other contributors who need to build the application would need some kind of solution (either being given a key to download directly, or we have a different secure location that they can download the files from)

**Pull request closed without merging**